.circleci/config.yml

```diff
@@ -17,6 +18,9 @@ workflows:
 
 version: 2.1
 
+orbs:
+  shellcheck: circleci/shellcheck@3.1.1
```

Comment:
Definitely! If there are many repeated tasks there that could also be reused by other repos in the organization, it would be beneficial to split them in orbs.

Reply:
There is some common stuff. For example the [gitter notification command](https://github.com/ethereum/solidity/blob/v0.8.17/.circleci/config.yml#L34-L93) may be a good candidate for an orb (while at it, we could also make them work on Windows). The [`t_ems_ext`](https://github.com/ethereum/solidity/blob/v0.8.17/.circleci/config.yml#L1210-L1271) is used by external tests and [their definitions](https://github.com/ethereum/solidity/blob/v0.8.17/.circleci/config.yml#L538-L650) are pretty repetitive too. And there are some common [steps/templates](https://github.com/ethereum/solidity/blob/v0.8.17/.circleci/config.yml#L118-L285). Also generally splitting the whole file into smaller bits by grouping similar jobs (build/test jobs, bytecode compare, external tests, etc.) would be great if possible.

We have a whole issue about that (https://github.com/ethereum/solidity/issues/11846) but there are limits to what we can do and so far I did not consider creating orbs.

Reply:
Nice! Thanks, I can take a look on that :)
I can start with the gitter notification then, since there is already an issue for the failing solc-bin nightly builds.